### ISSUE TITLE
[Help] Can't build following instructions in README

### ISSUE THREAD

**Issue by nefarius** [sharpdx/SharpDX#1076]
Greetings,

I tripple-checked that I've got all the required dependencies installed but I unfortunately can't build any of the projects. I guess I'm still missing vital dependencies which are not resolved:

```
NETSDK1004	Assets file '...\SharpDX\Source\SharpDX\obj\project.assets.json' not found. Run a NuGet package restore to generate this file.	SharpDX	C:\Program Files\dotnet\sdk\2.1.401\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets	198	
```

`build.cmd`, `msbuild /restore` and `dotnet restore` all fail (see screenshots).

![image](https://user-images.githubusercontent.com/286631/45240087-ae85e480-b2e7-11e8-86e1-cce45906c165.png)

![image](https://user-images.githubusercontent.com/286631/45240239-3b30a280-b2e8-11e8-8fe6-3361b1541634.png)

![image](https://user-images.githubusercontent.com/286631/45240075-a6c64000-b2e7-11e8-8b77-9d32fbeeabf9.png)

Help pls 🙂  🐱

**Comment by evilC** [comment]
Similar issues to Nefarius, seems to be some kind of problem with .Net Dependencies:  
![](https://i.imgur.com/YuqFHdl.png)

**Comment by xoofx** [comment]
@jkoritzinsky I think I had the same problem with the current codebase where I had to run the build manually to get the assemblies in order to compile more incrementally when opening the VS solutions. Could you improve the situation so that we can compile everything from VS as it was done before?

**Comment by jkoritzinsky** [comment]
I'll see what I can do.

**Comment by nefarius** [comment]
Ah, glad to hear it's not just my incompetence 😅

**Comment by Alan-FGR** [comment]
Hey guys, are the generated pinvokes (e.g. D3D11 stuff) available somewhere?
I see the generated pinvokes for D3D10 are in the repo but the others aren't.
Thank you.

**Comment by jkoritzinsky** [comment]
They're generated as part of the build.

**Comment by Alan-FGR** [comment]
Yes, but since I can't build I was wondering if they're available somewhere else so I can try and add them manually since I don't really need the whole DirectX, just Direct3D11 (and deps). I thought that maybe they'd be available in appveyor but I couldn't find a way to get that.

**Comment by bdachev** [comment]
Out of curiosity I tried to compile latest SharpDX from master and I succeeded eventually but had following problems:
 - it fails at ParseCPlusPlus task if there are spaces in project path - probably a bug in SharpGenTools not quoting something
 - Debug build of SharpDX project fails if Release is not build before

**Comment by jkoritzinsky** [comment]
The issue in the `ParseCPlusPlus` task is a bug in SharpGenTools. Can you file an issue there?

**Comment by bdachev** [comment]
Hi Jeremy, you are correct that I don't have 10.0.14393 SDK installed.
What worried me was that it was able to build Debug from scratch without that SDK but failed on Release so maybe there is something simple that needs to be fixed for that configuration.
Anyway I can confirm that can build both Debug and Release configurations from scratch so it seems your latest contributions have fixed it. And also I can confirm that it can be build without 10.0.14393 SDK so maybe it is no longer needed. I do have installed 10.0.16299 though (both desktop and UWP).
B.

**Comment by bdachev** [comment]
OK. My mistake. It turned out that I have 10.0.14393 SDK installed on my system kept by VS2015. When I uninistalled it SharpDX no longer builds. So it is still a requirement.
